mlflow/gateway/providers/palm.py

```diff
@@ -0,0 +1,189 @@
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        key_mapping = {
+            "stop": "stopSequences",
+            "candidate_count": "candidateCount",
+            "max_tokens": "maxOutputTokens",
+        }
+        for k1, k2 in key_mapping.items():
+            if k2 in payload:
+                raise HTTPException(
+                    status_code=422, detail=f"Invalid parameter {k2}. Use {k1} instead."
+                )
+        payload = rename_payload_keys(payload, key_mapping)
+
+        # Replace 'role' with 'author' in payload
+        for m in payload["messages"]:
+            if "role" in m:
+                m["author"] = m.pop("role")
+
+        # Map 'messages', 'examples, and 'context' to 'prompt'
+        prompt = {"messages": payload.pop("messages")}
+        if "examples" in payload:
+            prompt["examples"] = payload.pop("examples")
+        if "context" in payload:
+            prompt["context"] = payload.pop("context")
+        payload["prompt"] = prompt
+
+        resp = await self._request(
+            f"{self.config.model.name}:generateMessage",
+            payload,
+        )
+
+        # Response example
+        # (https://developers.generativeai.google/api/rest/generativelanguage/models/generateMessage)
+        # ```
+        # {
+        #   "candidates": [
+        #     {
+        #       "author": "1",
+        #       "content": "Hi there! How can I help you today?"
+        #     }
+        #   ],
+        #   "messages": [
+        #     {
+        #       "author": "0",
+        #       "content": "hi"
+        #     }
+        #   ]
+        # }
+        # ```
+        return chat.ResponsePayload(
+            **{
+                "candidates": [
+                    {
+                        "message": {
+                            "role": c["author"],
+                            "content": c["content"],
+                        },
+                        "metadata": {},
+                    }
+                    for c in resp["candidates"]
+                ],
+                "metadata": {
+                    "model": self.config.model.name,
+                    "route_type": self.config.route_type,
+                },
+            }
+        )
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        key_mapping = {
+            "stop": "stopSequences",
+            "candidate_count": "candidateCount",
+            "max_tokens": "maxOutputTokens",
+        }
+        for k1, k2 in key_mapping.items():
+            if k2 in payload:
+                raise HTTPException(
+                    status_code=400, detail=f"Invalid parameter {k2}. Use {k1} instead."
+                )
```

Comment:
`chat` uses 422. Can we use the same status code?

Reply:
+1 422 is the correct error code

Reply:
Updated all of the other providers to match the error code as well